concent_api/core/tests/test_unit_model.py

```diff
@@ -162,3 +170,243 @@ def test_that_storing_subtask_with_task_to_compute_nested_in_another_messages_wi
                 force_get_task_result=force_get_task_result,
                 subtask_results_rejected=subtask_results_rejected,
             )
+
+
+class TestDepositAccountValidation(ConcentIntegrationTestCase):
+    (PROVIDER_PRIVATE_KEY, PROVIDER_PUBLIC_KEY) = generate_ecc_key_pair()
+
+    def generate_ethereum_public_key(self):  # pylint: disable=no-self-use
+        ecc = ECCx(None)
+        return encode_hex(privtoaddr(ecc.raw_privkey))
+
+    def store_report_computed_task_as_subtask(self):  # pylint: disable=no-self-use
+        task_to_compute = factories.tasks.TaskToComputeFactory()
+        report_computed_task = factories.tasks.ReportComputedTaskFactory(task_to_compute=task_to_compute)
+        ack_report_computed_task = factories.tasks.AckReportComputedTaskFactory(report_computed_task=report_computed_task)
+        force_get_task_result = factories.concents.ForceGetTaskResultFactory(report_computed_task=report_computed_task)
+        subtask_results_rejected = factories.tasks.SubtaskResultsRejectedFactory(report_computed_task=report_computed_task)
+
+        store_subtask(
+            task_id=task_to_compute.task_id,
+            subtask_id=task_to_compute.subtask_id,
+            provider_public_key=hex_to_bytes_convert(task_to_compute.provider_public_key),
+            requestor_public_key=hex_to_bytes_convert(task_to_compute.requestor_public_key),
+            state=Subtask.SubtaskState.ACCEPTED,
+            task_to_compute=task_to_compute,
+            report_computed_task=report_computed_task,
+            next_deadline=None,
+            ack_report_computed_task=ack_report_computed_task,
+            reject_report_computed_task=None,
+            force_get_task_result=force_get_task_result,
+            subtask_results_rejected=subtask_results_rejected,
+        )
+
+    def test_that_exception_is_rised_when_ethereum_public_key_is_not_equal_42_characters(self):
+        ethereum_public_key = self.generate_ethereum_public_key()
+        DA = DepositAccount()
```

Comment:
Might be painful but we try to use full names, I guess in tests too. @dybi what do you think? Can it stay this way?

Reply:
Done.